src/server/dflycmd.cc

```diff
@@ -202,12 +144,114 @@ void DflyCmd::HandleJournal(CmdArgList args, ConnectionContext* cntx) {
   return rb->SendError(reply, kSyntaxErrType);
 }
 
-uint32_t DflyCmd::AllocateSyncSession() {
+void DflyCmd::Thread(CmdArgList args, ConnectionContext* cntx) {
+  RedisReplyBuilder* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  util::ProactorPool* pool = shard_set->pool();
+
+  if (args.size() == 2) {  // DFLY THREAD : returns connection thread index and number of threads.
+    rb->StartArray(2);
+    rb->SendLong(ProactorBase::GetIndex());
+    rb->SendLong(long(pool->size()));
+    return;
+  }
+
+  // DFLY THREAD to_thread : migrates current connection to a different thread.
+  string_view arg = ArgS(args, 2);
+  unsigned num_thread;
+  if (!absl::SimpleAtoi(arg, &num_thread)) {
+    return rb->SendError(kSyntaxErr);
+  }
+
+  if (num_thread < pool->size()) {
+    if (int(num_thread) != ProactorBase::GetIndex()) {
+      listener_->Migrate(cntx->owner(), pool->at(num_thread));
+    }
+
+    return rb->SendOk();
+  }
+
+  rb->SendError(kInvalidIntErr);
+  return;
+}
+
+void DflyCmd::Sync(CmdArgList args, ConnectionContext* cntx) {
```

Comment:
please add a comment what this functions does / intended to do. I frankly do not remember now.